src/transform/tests/testdata/partial-reduction-pushdown

```diff
@@ -0,0 +1,498 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Tests for partial reduction pushdown to constant inputs.
+#
+# TODO: Implement support for partial reduction pushdown.
+# The general idea was discussed in
+# https://github.com/MaterializeInc/materialize/issues/10119, but we
+# decided that we need to spend more time on formalizing the proposed
+# approach, to ensure it is actually correct. Until we have done so,
+# the tests here only exercise the non-partial reduction pushdown
+# optimizationand are mostly equivalent to the tests in
+# `reduction-pushdown`.
+
+cat
+(defsource w [int16 int32])
+(defsource x [int32 string])
+(defsource y ([int16 string] [[1]]))
+(defsource z [int16 string])
+----
+ok
+
+# Distinct Pushdown tests
+
+build apply=ReductionPushdown
+(reduce (join [(get x) (constant [] [int32 string])] [[#1 #3]]) [#1] [])
+----
+----
+%0 =
```

Comment:
Are these tests defining the current or the intended behavior?

Reply:
The current behavior. AFAICT, there is no way to specify tests that are expected to fail.